src/newmagic.cpp

```diff
@@ -1708,6 +1708,39 @@ void cast_health_spell(struct char_data *ch, int spell, int sub, int force, char
         send_to_char(FAILED_CAST, ch);
       spell_drain(ch, spell, force, 0);
       break;
+    case SPELL_NIGHTVISION:
+      WAIT_STATE(ch, (int) (SPELL_WAIT_STATE_TIME));
+      success = success_test(skill, 6 + target_modifiers);
+      if (success > 0 && !AFF_FLAGGED(ch, AFF_LOW_LIGHT)) {
+        create_sustained(ch, vict, spell, force, 0, success, spells[spell].draindamage);
+        send_to_char("Your eyes tingle as the shadows around you become clearer.\r\n", vict);
+        act("You successfully sustain that spell on $N.", FALSE, ch, 0, vict, TO_CHAR);
+      } else
+        send_to_char(FAILED_CAST, ch);
+      spell_drain(ch, spell, force, 0);
+      break;
+    case SPELL_INFRAVISION:
+      WAIT_STATE(ch, (int) (SPELL_WAIT_STATE_TIME));
+      success = success_test(skill, 6 + target_modifiers);
+      if (success > 0 && !AFF_FLAGGED(ch, AFF_INFRAVISION)) {
```

Comment:
IIRC you'll have to write in logic to apply / remove the aff_infra flag when the spell is cast and removed, I don't actually see anything that adds the flag (unless I just missed it?)